openstates/in/committees.py

```diff
@@ -0,0 +1,115 @@
+            person = mem["firstName"]+" "+mem["lastName"]
+            comm.add_member(person,role=role_name)
+            return person
+        return None
+
+
+    def get_subcommittee_info(self,session):
+        #api gives NO way of finding out who owns
+        #a subcommittee. It can be found based in indenting(!)
+        #here: http://iga.in.gov/legislative/2015/committees/standing
+        #so we're going to hit that and make a dictionary. yuck
+
+        #but this is less important than some other stuff
+        #so we're going to be OK if we timeout.
+        link = "http://iga.in.gov/legislative/{}/committees/standing".format(session)
+        html = get_with_increasing_timeout(self,link,fail=False)
+        sc_dict = {}
+        if html:
+            doc = lxml.html.fromstring(html.text)
+            committees = doc.xpath("//li[@class='committee-item']")
+            for c in committees:
+                comm_name = c.xpath("./a")[0]
+                comm_name = comm_name.text_content().strip()
+                subcomms = c.xpath(".//li[@class='subcommittee-item']")
+                for s in subcomms:
+                    subcom_name = s.text_content().strip()
+                    sc_dict[subcom_name] = comm_name
+
+
+        return sc_dict
+        
+
+    def scrape(self,term,chambers):
+        t = next((item for item in self.metadata["terms"] if item["name"] == term),None)
+        session = max(t["sessions"])
+
+        subcomms = self.get_subcommittee_info(session)
+
+        api_base_url = "https://api.iga.in.gov/{}/committees/".format(session)
+        html_base_url = "http://iga.in.gov/legislative/{}/committees/".format(session)
+        client = ApiClient(self)
+        r = client.get("committees",session=session)
+        all_pages = client.unpaginate(r)
+        for comm_info in all_pages:
+            #this is kind of roundabout, but needed in order
+            #to take advantage of all of our machinery to make
+            #sure we're not overloading their api
+            comm_link = comm_info["link"]
+            comm_name = comm_link.split("/")[-1]
+            comm_json = client.get("committee",session=session,committee_name=comm_name)
+            
+            chamber = comm_json["chamber"]["name"]
+            if chamber == "Senate":
+                chamber = "upper"
+            elif chamber == "House":
+                chamber = "lower"
+            else:
+                chamber = "joint"
+
+            name = comm_json["name"]
+            try:
+                owning_comm = subcomms[name]
+            except KeyError:
+                comm = Committee(chamber,name)
+            else:
+                comm = Committee(chamber,owning_comm,subcommittee=name)
+
+            chair = self.process_special_members(comm,comm_json,"chair")
+            vicechair = self.process_special_members(comm,comm_json,"viceChair")
+            ranking = self.process_special_members(comm,comm_json,"rankingMinMember")
+
```

Comment:
I can't think of a way to catch errors here since they're fieldnames in the JSON, so we have to go looking for them specifically (I mean, I could loop through all JSON fieldnames and look for a person's name, but that feels like more risk than reward because who knows what else could show up).

Oh, also also, the positions seem to be null when not filled, so it's unlikely that there will be weird ones on just one committee.